jetty9/src/main/java/com/thoughtworks/go/server/Jetty9Server.java

```diff
@@ -59,38 +60,42 @@
     private WebAppContext webAppContext;
     private static final Logger LOG = LoggerFactory.getLogger(Jetty9Server.class);
     private GoSSLConfig goSSLConfig;
+    private final DeploymentManager deploymentManager;
 
     public Jetty9Server(SystemEnvironment systemEnvironment, String password, SSLSocketFactory sslSocketFactory) {
-        this(systemEnvironment, password, sslSocketFactory, new Server());
+        this(systemEnvironment, password, sslSocketFactory, new Server(), new DeploymentManager());
     }
 
-    Jetty9Server(SystemEnvironment systemEnvironment, String password, SSLSocketFactory sslSocketFactory, Server server) {
+    Jetty9Server(SystemEnvironment systemEnvironment, String password, SSLSocketFactory sslSocketFactory, Server server, DeploymentManager deploymentManager) {
         super(systemEnvironment, password, sslSocketFactory);
         systemEnvironment.set(SystemEnvironment.JETTY_XML_FILE_NAME, JETTY_XML);
         this.server = server;
         goSSLConfig = new GoSSLConfig(systemEnvironment);
+        this.deploymentManager = deploymentManager;
     }
 
     @Override
     public void configure() throws Exception {
         server.addEventListener(mbeans());
         server.addConnector(plainConnector());
         server.addConnector(sslConnector());
-        HandlerCollection handlers = new HandlerCollection();
-        handlers.addHandler(welcomeFileHandler());
+        ContextHandlerCollection handlers = new ContextHandlerCollection();
```

Comment:
This is what changes how `/` and `/go` etc. are handled. Earlier, each handler was checked and the one that handled it won. Now, based on `/` or `/go` or `/go/assets`, the best handler is picked and used. No difference in behavior.